### PR TITLE
[AOTI] Fix bfloat16 in CPU

### LINKED ISSUE #122986: [AOTI] error: ‘bfloat16’  was not declared in this scope
### 🐛 Describe the bug

How to repro:

comment out the line below:

https://github.com/pytorch/pytorch/blob/57a9a64e10b84fa8d932482ae9417aa3fc3fbf44/test/inductor/test_aot_inductor.py#L2315

and run the following command:

```
$ python test/inductor/test_aot_inductor.py -k test_sdpa_abi_compatible_cpu_with_stack_allocation_and_minimal_arrayref_interface
```

### Versions

PyTorch version: 2.3.0a0+git57a9a6
Is debug build: True
CUDA used to build PyTorch: 12.0
ROCM used to build PyTorch: N/A

cc @ezyang @msaroufim @bdhirsh @anijain2305 @zou3519 @chauhang @desertfire

### PR DESCRIPTION
**Pull request by yushangdi**
Fixes #122986

- add "typedef at::BFloat16 bfloat16;" to the header of generated cpp file

- Supress warning: comparison of integer expressions of different signedness: ‘long unsigned int’ and ‘int64_t’ {aka ‘long int’} [-Wsign-compare]
  436 |   if (tensor.numel() != numel) {

cc @voznesenskym @penguinwu @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @wenzhe-nrv @jiayisunx @ipiszy @yf225 @chenyang78 @kadeng @muchulee8 @ColinPeppler @amjames @desertfire @chauhang

### CODE REVIEW

**Review by chenyang78 — APPROVED** [review]
LGTM. Thanks!

File: torch/_inductor/codegen/cpp_wrapper_cpu.py [hunk 1]
```diff
@@ -171,6 +171,7 @@ def write_header(self):
                     #include <torch/csrc/inductor/aoti_runtime/model.h>
                     """
                 )
+            self.header.splice("typedef at::BFloat16 bfloat16;")
```
Comment (desertfire) [inline torch/_inductor/codegen/cpp_wrapper_cpu.py:174]: Nit: move https://github.com/pytorch/pytorch/blob/f4f7aba75d6e34d090e7089bb95d608b6ac90731/torch/_inductor/codegen/cpp_wrapper_cuda.py#L46-L47 here.

**Review by desertfire — APPROVED** [review]

**Pull request closed without merging**